packages/schema/src/config/typescript.ts

```diff
@@ -13,6 +14,14 @@ export default defineUntypedSchema({
      */
     strict: true,
 
+    /**
+     * whether or not to write declaration files to add specific types based on your builder environment. It is
+     * advised to turn this off for module authors, who should support multiple possible builders.
+     */
+    builderEnv: {
+      $resolve: async (val, get) => val ?? await resolvePath('@nuxt/module-builder', { url: await get('modulesDir') }).then(path => !path).catch(() => null) ?? true,
```

Comment:
If both are installed in a mono-repo, this can make issues. Can we simply support `typescript.builder` as a way to override default (top level) `builder` only for types and also use `vite | webpack` instead of package names?

Reply:
I'm also very happy to add the override directly from nuxt/module-builder. 👍